### PR TITLE
Deprecate `line_terminator` in favor of `lineterminator` in `to_csv`

### LINKED ISSUE #12894: [BUG] API mismatch - DataFrame.to_csv line_terminator vs pandas' lineterminator
**Describe the bug**
writing code with `import cudf as pd`

**Steps/Code to reproduce bug**

[pandas to_csv uses `lineterminator`](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.to_csv.html#pandas.DataFrame.to_csv)

[cudf to_csv uses `line_terminator`](https://docs.rapids.ai/api/cudf/stable/api_docs/api/cudf.dataframe.to_csv#cudf.DataFrame.to_csv)

**Expected behavior**
`cudf to_csv` to accept `lineterminator`

https://github.com/rapidsai/cudf/blob/branch-23.02/python/cudf/cudf/io/csv.py#L158

### PR DESCRIPTION
**Pull request by wence-**
## Description
Pandas 1.5 deprecated line_terminator in favour of lineterminator (to align with read_csv), and removed it in 2.0. Align the cuDF API by preferring lineterminator and providing a deprecation warning for use of line_terminator. Closes #12894.

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

File: python/cudf/cudf/core/dataframe.py [hunk 1]
```diff
@@ -6358,13 +6356,29 @@ def to_csv(
...
+        if line_terminator is not None:
+            warnings.warn(
+                "line_terminator is a deprecated keyword argument, "
+                "use lineterminator instead.",
+                FutureWarning,
+            )
+            if lineterminator is not None:
+                warnings.warn(
+                    f"Ignoring {line_terminator=} in favour "
+                    f"of {lineterminator=}"
```
Comment (galipremsagar) [inline python/cudf/cudf/core/dataframe.py:6376]: Nice, TIL about this f-string feature.

**Comment by galipremsagar** [comment]
@wence-  Just FYI since I couldn't suggest from GitHub UI, I also made a trivial change to fix formatting: https://github.com/rapidsai/cudf/pull/12896/commits/fb0f7a27d2eb88717ddeaad1818fccf1864db43c

File: python/cudf/cudf/utils/ioutils.py [hunk 2]
```diff
@@ -1226,7 +1226,15 @@
 compression : str, None
     A string representing the compression scheme to use in the the output file
     Compression while writing csv is not supported currently
-line_terminator : char, default '\\n'
+line_terminator : str, optional
+
+    .. deprecated:: 23.04
+
+        Replaced with lineterminator for consistency with read_csv and pandas.
```
[the remark this answers is not in the document]
Comment (wence-) [reply python/cudf/cudf/utils/ioutils.py:1233]: Do you actually mean that (italics)? I should probably link read_csv and pandas.to_csv.

[the remark this answers is not in the document]
Comment (galipremsagar) [reply python/cudf/cudf/utils/ioutils.py:1233]: Yeah, much better to cross-link here. But if pandas.to_csv linking is more complex here, just '``' would do

File: docs/cudf/source/conf.py [hunk 3]
```diff
@@ -201,6 +201,7 @@
     "cupy": ("https://docs.cupy.dev/en/stable/", None),
     "numpy": ("https://numpy.org/doc/stable", None),
     "pyarrow": ("https://arrow.apache.org/docs/", None),
+    "pandas": ("https://pandas.pydata.org/docs/", None),
```
Comment (galipremsagar) [inline docs/cudf/source/conf.py:204]: Awesome 🎉

File: python/cudf/cudf/core/dataframe.py [hunk 4]
```diff
@@ -6358,13 +6356,29 @@ def to_csv(
...
 
+        if line_terminator is not None:
+            warnings.warn(
+                "line_terminator is a deprecated keyword argument, "
+                "use lineterminator instead.",
+                FutureWarning,
+            )
+            if lineterminator is not None:
+                warnings.warn(
+                    f"Ignoring {line_terminator=} in favour "
```
Comment (bdice) [inline python/cudf/cudf/core/dataframe.py:6375]: US spelling.
```suggestion
                    f"Ignoring {line_terminator=} in favor "
```

**Review by bdice — APPROVED** [review]

**Pull request merged**